shared/lib/keyring.ts

```diff
@@ -0,0 +1,43 @@
+import {
+  KeyringObject,
+  KeyringMetadata,
+  KeyringTypes,
+} from '@metamask/keyring-controller';
+
+// TODO: This kind of logic should be inside the `KeyringController` (using `KeyringSelector` query, or make `addNewKeyring` returns it keyring ID alongside
+
+export function findKeyringId(
+  keyrings: KeyringObject[],
+  keyringsMetadata: KeyringMetadata[],
+  params: { address?: string; type?: KeyringTypes },
+): string {
+  const keyringIndex = keyrings.findIndex((keyring) => {
+    if (params.address && params.type) {
+      return (
+        keyring.accounts.includes(params.address) &&
+        keyring.type === params.type
+      );
+    }
+    if (params.address) {
+      return keyring.accounts.includes(params.address.toLowerCase());
+    }
+    if (params.type) {
+      return keyring.type === params.type;
+    }
+    throw new Error('Must provide either address or type parameter');
+  });
```

Comment:
Cosmetic: Just use `selector` (similar to `KeyringController.withKeyring` method)

And also, we missed a `.toLowerCase()` in one of the branch.

```suggestion
  selector: { address?: string; type?: KeyringTypes },
): string {
  const keyringIndex = keyrings.findIndex((keyring) => {
    if (selector.address && selector.type) {
      return (
        keyring.accounts.includes(selector.address.toLowerCase()) &&
        keyring.type === selector.type
      );
    }
    if (selector.address) {
      return keyring.accounts.includes(selector.address.toLowerCase());
    }
    if (selector.type) {
      return keyring.type === selector.type;
    }
    
    throw new Error('Must provide either address or type selector');
  });
```

I tried to refactor to avoid the repetition, but I feel like the code is quite readable as it is now. 😄